File: app/src/main/java/com/pij/zworkout/uc/WorkoutPersistenceUC.java
package com.pij.zworkout.uc;

import com.annimon.stream.Optional;
import com.pij.zworkout.service.api.WorkoutFile;

import java.io.File;
import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;

/**
 * <p>Created on 02/03/2018.</p>
 *
 * @author Pierrejean
 */

public interface WorkoutPersistenceUC {

    Observable<List<WorkoutFile>> workouts();

    // TODO replace File by URI
    Single<File> save(Workout data, Optional<File> file);

    // TODO replace File by URI
    Single<Workout> load(File source);
}

File: app/src/main/java/com/pij/zworkout/uc/WorkoutPersistenceUC.kt
package com.pij.zworkout.uc

import com.annimon.stream.Optional
import com.pij.zworkout.service.api.WorkoutFile
import io.reactivex.Observable
import io.reactivex.Single
import java.io.File

/**
 *
 * Created on 02/03/2018.
 *
 * @author Pierrejean
 */

interface WorkoutPersistenceUC {

    fun workouts(): Observable<List<WorkoutFile>>

    // TODO replace File by URI
    fun save(data: Workout, target: Optional<File>): Single<File>

    // TODO replace File by URI
    fun load(source: File): Single<Workout>
}
